numpy/f2py/tests/test_array_from_pyobj.py skips all tests due to use of legacy dtype API

There is a try/except in the f2py util testing code that masks compilation errors:

https://github.com/numpy/numpy/blob/abeca76c5ca956d4bfa27e01fa4988c211ba8444/numpy/f2py/tests/util.py#L329-L332

If I delete that, all the tests in `test_array_from_pyobj.py` fail with the following compilation error:

```
/Users/goldbaum/Documents/numpy/build-install/usr/lib/python3.13/site-packages/numpy/f2py/tests/src/array_from_pyobj/wrapmodule.c:118:44: error: no member named 'alignment' in 'struct _PyArray_Descr'
                       PyArray_DESCR(arr)->alignment,
                       ~~~~~~~~~~~~~~~~~~  ^
```

The try/except seems like a bad idea but I guess it's there for a reason for distutils support. Probably worth seeing if there are other tests like this.

--- a/numpy/f2py/tests/src/array_from_pyobj/wrapmodule.c
+++ b/numpy/f2py/tests/src/array_from_pyobj/wrapmodule.c
@@ -115,7 +115,7 @@ static PyObject *f2py_rout_wrap_attrs(PyObject *capi_self,
                        PyArray_DESCR(arr)->type,
                        PyArray_TYPE(arr),
                        PyArray_ITEMSIZE(arr),
-                       PyArray_DESCR(arr)->alignment,
+                       PyDataType_ALIGNMENT(arr),
                        PyArray_FLAGS(arr),
                        PyArray_ITEMSIZE(arr));
 }

--- a/numpy/f2py/tests/util.py
+++ b/numpy/f2py/tests/util.py
@@ -328,7 +328,7 @@ def build_meson(source_files, module_name=None, **kwargs):
     # compiler stack is on the CI
     try:
         backend.compile()
-    except:
+    except subprocess.CalledProcessError:
         pytest.skip("Failed to compile module")
 
     # Import the compiled module

